lib/pos_blockchain.go

```diff
@@ -293,8 +293,31 @@ func (bc *Blockchain) validateBlockView(desoBlock *MsgDeSoBlock) error {
 // validateBlockLeader validates that the proposer is the expected proposer for the
 // block height + view number pair.
 func (bc *Blockchain) validateBlockLeader(desoBlock *MsgDeSoBlock) error {
-	// TODO: Implement me
-	return errors.New("IMPLEMENT ME")
+	utxoView, err := NewUtxoView(bc.db, bc.params, bc.postgres, bc.snapshot)
+	if err != nil {
+		return errors.Wrapf(err, "validateBlockLeader: Problem initializing UtxoView")
+	}
+	currentEpochEntry, err := utxoView.GetCurrentEpochEntry()
+	if err != nil {
+		return errors.Wrapf(err, "validateBlockLeader: Problem getting current epoch entry")
+	}
+	leaders, err := utxoView.GetSnapshotLeaderSchedule()
+	if err != nil {
+		return errors.Wrapf(err, "validateBlockLeader: Problem getting leader schedule")
+	}
+	// TODO: Do I need to validate the views + heights before doing the subtraction.
```

Comment:
It doesn't hurt to validate that `desoBlock.Header.ProposedInView-currentEpochEntry.InitialView >= desoBlock.Header.Height-currentEpochEntry.InitialBlockHeight`, and that `len(leaders) > 0`. Although I can't imagine when either of these validations will fail.

Beefing up the comment on why we use this formula is also good. Basically, we the formula tells us how many times we've timed out during this epoch. This tells us how far along the leader schedule we are.